swagger.yaml

```diff
@@ -1,42 +1,254 @@
+          schema:
+            $ref: "#/definitions/Job"
+        500:
+          $ref: '#/responses/InternalError'
+        404:
+          $ref: '#/responses/ResourceNotFound'
+
+  /jobs:
+    get:
+      tags:
+        - "Public"
+      summary: Get a list of jobs.
+      description: Get a list of search reindex jobs that meet the criteria specified by the four query parameters.
+      parameters:
+        - $ref: '#/parameters/limit'
+        - $ref: '#/parameters/offset'
+        - $ref: '#/parameters/sort'
+        - $ref: '#/parameters/state'
+      produces:
+        - application/json
+      responses:
+        200:
+          description: Successfully returned the requested list of jobs.
+          schema:
+            $ref: "#/definitions/Jobs"
+        500:
+          $ref: '#/responses/InternalError'
+        400:
+          $ref: '#/responses/InvalidRequest'
+    post:
+      tags:
+        - "Public"
+      summary: Create a new search reindex job.
+      description: Post details of a new search reindex job and receive the new Job object created in response.
+      parameters:
+        - $ref: '#/parameters/job'
       produces:
         - application/json
       responses:
         200:
           description: OK
           schema:
-            $ref: "#/definitions/HelloResponse"
+            $ref: "#/definitions/Job"
         500:
           $ref: '#/responses/InternalError'
 
 responses:
+
   InternalError:
     description: "Failed to process the request due to an internal error"
 
+  ResourceNotFound:
+    description: "Failed to locate a Job that has the specified id"
+
+  InvalidRequest:
+    description: "The request does not contain the correct parameters and/or values in its parameters"
+
 definitions:
-  HelloResponse:
+
+  Job:
+    type: object
+    required:
+      - "id"
+      - "last_updated"
+      - "links"
+      - "search_index_name"
+      - "state"
+    properties:
+      id:
+        type: string
+      last_updated:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      links:
+        type: object
+        required:
+          - "tasks"
+          - "self"
+        properties:
+          tasks:
+            type: string
+            format: http://localhost:<port>/jobs/{id}/tasks
+          self:
+            type: string
+            format: http://localhost:<port>/jobs/{id}
+      number_of_tasks:
+        type: integer
+      reindex_completed:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      reindex_failed:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      reindex_started:
+        type: string
+        format: date
+        description: ISO8601 timestamp
+      search_index_name:
+        type: string
+      state:
+        type: string
+        enum:
+          - CREATED
+          - IN-PROGRESS
+          - COMPLETED
+          - FAILED
+      total_search_documents:
```

Comment:
Every field should contain a description. This should be very explicit, see other swagger specs such as the [dataset API](https://github.com/ONSdigital/dp-dataset-api/blob/develop/swagger.yaml) or [recipe API](https://github.com/ONSdigital/dp-recipe-api/blob/develop/swagger.yaml).

Reply:
The reason for being explicit is that it helps explain what the fields are, usually we use the swagger yaml file to generate documentation for our APIs on the developer site. Due to this application being solely private it wont be publicised in the same manner as some of our other APIs.

Reply:
descriptions now added - it would be good to know how to generate documentation from the swagger yaml